### ISSUE TITLE
Can't reschedule a missed appointment

### ISSUE THREAD

**Issue by PeerRich** [calcom/cal.com#15640]
### Discussed in https://github.com/calcom/cal.com/discussions/15620

<div type='discussions-op-text'>

<sup>Originally posted by **animoller** June 29, 2024</sup>
Gidday, I have a prospective client who missed a discovery call. I asked them to reschedule rather than rebook as they included a good chunk of information in the registration page about what they're looking for. It appears to let you reschedule (choose a new day/time) but you get an error that is completely useless from a CX perspective "invalid string". 

The first name, last name and email address are greyed out. They did include a last name when they booked but it's not showing up in the reschedule view. Is this the invalid string, or is it because you can't reschedule an event that's passed?

Cheers

![required field bug](https://github.com/calcom/cal.com/assets/174170958/afea68fe-ffff-4378-a386-970c515fa566)
</div>

**Comment by theonly1me** [comment]
@PeerRich I'm looking into this, can you please assign it to me? Thanks.

I've also found a couple of other issues related to this workflow, will create the GitHub issues soon.

**Comment by Amit91848** [comment]
> @PeerRich I'm looking into this, can you please assign it to me? Thanks.
> 
> I've also found a couple of other issues related to this workflow, will create the GitHub issues soon.

Hey @theonly1me! We don't assign issues. You can create a PR and we will surely look into it.

**Comment by Pushpendra24-7** [comment]
hey i want to work on this issue

**Comment by Goul-duKat** [comment]
Good day

We can not reschedule at all !
![Immagine 2024-07-03 152211](https://github.com/calcom/cal.com/assets/174621369/1d7ffca4-5cbb-4c51-9730-258a910f937b)

We are using all anonymous incoming users, so they don't need and will be not log ins in the system just managed on the single appointment for the personal information like email.

We started using cal.com with version  4.1.12 and now we updated to the 4.2.3, we seen this :
fix: Seat Attendee Rescheduling Logic by @joeauyeung in https://github.com/calcom/cal.com/pull/14784

But it was broken and still broken on last version.

Someone can take a look why we get the blue text and impossible to reschedule ?

Thanks
Cheers

**Comment by theonly1me** [comment]
#15661 fixes this issue.

Here's how to replicate the issue:
- Visit http://cal.com/event-types/1135?tabName=advanced 
- Edit the name field to split into "First Name" and "Last Name"
- Set the "First Name" as required and "Last Name" as not required
- Go to the booking link in an incoginto tab
- Fill in the First Name, leave out the last name and make the booking
- Once the booking is done, update the setting for "Last Name" to required
- Go back to the booking page and try to reschedule
- Since the name fields are un-editable, it shows a validation error

Before Fix: 

https://github.com/calcom/cal.com/assets/19223383/e3987987-3573-4ff0-86c1-15d02e369ccf

To fix this, 
- We check if this is a "reschedule" and not a new booking.
- If the user is rescheduling, we check if the variant is "firstAndLastName".
- If it is, we verify if the last name is required. If the last name is required and is empty:
- We set both the first and last names to empty, so the user can add them both during the booking.
- The reason we need to set both to empty is due to the onChange handler being passed into `react-hook-form`. If we don't set both to empty, it replaces the value with an object like { lastName: "name" }, causing the Zod validations to fail.

After Fix:

https://github.com/calcom/cal.com/assets/19223383/8bc9dca6-7858-4275-8c0f-1a2e3b5d2091

**Comment by theonly1me** [comment]
> Good day
> 
> We can not reschedule at all ! ![Immagine 2024-07-03 152211](https://private-user-images.githubusercontent.com/174621369/345444459-1d7ffca4-5cbb-4c51-9730-258a910f937b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNDI2MTksIm5iZiI6MTcyMDA0MjMxOSwicGF0aCI6Ii8xNzQ2MjEzNjkvMzQ1NDQ0NDU5LTFkN2ZmY2E0LTVjYmItNGM1MS05NzMwLTI1OGE5MTBmOTM3Yi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwM1QyMTMxNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNzc5ZGRiYzhhZGJkNWZlZDhlYmVmNDMxMjg3MGM4YTVjYTE0ZTFhZDI4ZjJjYWE5NTZlN2QwMGFiNDZmOTNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.O6eTC_XplRYYEInsqjWO4H8lTFQovGnkslXyW4FyeXY)
> 
> We are using all anonymous incoming users, so they don't need and will be not log ins in the system just managed on the single appointment for the personal information like email.
> 
> We started using cal.com with version 4.1.12 and now we updated to the 4.2.3, we seen this : fix: Seat Attendee Rescheduling Logic by @joeauyeung in #14784
> 
> But it was broken and still broken on last version.
> 
> Someone can take a look why we get the blue text and impossible to reschedule ?
> 
> Thanks Cheers

@Goul-duKat this is an unrelated issue, could you please create a new issue for this? Thanks!

**Comment by theonly1me** [comment]
> Good day
> 
> We can not reschedule at all ! ![Immagine 2024-07-03 152211](https://private-user-images.githubusercontent.com/174621369/345444459-1d7ffca4-5cbb-4c51-9730-258a910f937b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNzkyNzEsIm5iZiI6MTcyMDA3ODk3MSwicGF0aCI6Ii8xNzQ2MjEzNjkvMzQ1NDQ0NDU5LTFkN2ZmY2E0LTVjYmItNGM1MS05NzMwLTI1OGE5MTBmOTM3Yi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNFQwNzQyNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YjYxMGFkZmFiODE1ZTU0OTA1YWQ5ODczMDU3NTA4M2M2OTk4NWM3MTQ2MTZiZWRjZTIwNmI3ZDIwZmEyOGUyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GKQWO7YpMefAVDaOUpMFFg4G_gm5n8NbmSvolelnLEI)
> 
> We are using all anonymous incoming users, so they don't need and will be not log ins in the system just managed on the single appointment for the personal information like email.
> 
> We started using cal.com with version 4.1.12 and now we updated to the 4.2.3, we seen this : fix: Seat Attendee Rescheduling Logic by @joeauyeung in #14784
> 
> But it was broken and still broken on last version.
> 
> Someone can take a look why we get the blue text and impossible to reschedule ?
> 
> Thanks Cheers

@Goul-duKat this is due to a different, un-related bug, I've therefore created issue #15665 on your behalf. Please feel free to add any additional context via comments on #15665.

**Comment by theonly1me** [comment]
@PeerRich @Amit91848 can you please take a look at my responses on #15661? Please let me know if you think further changes are needed.

**Comment by CarinaWolli** [comment]
I haven't seen any recent issues about rescheduling problems. So closing this
